File: heap/KthLargestElement.java

import java.util.PriorityQueue;

public class KthLargestElement {
  public static void main(String[] args){
    Solution s = new Solution();
    System.out.println(s.findKthLargest(new int[]{3,2,1,5,6,4}, 2));
  }
}

class Solution{
  public int findKthLargest(int[] nums, int k){
    PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> b-a);
    for(int num : nums) pq.add(num);
    int result = -1;
    for(int i = 0; i < k; ++i) result = pq.poll();
    return result;
  }
}

File: cpp/kth_largest_element.cpp
#include <iostream>
#include <vector>
#include <queue>

using namespace std;

class Solution{
  public:
    int findKthLargest(vector<int>& nums, int k){
      priority_queue<int> pq;
      for(int i = 0; i < nums.size(); ++i){
        pq.push(nums[i]);
      }
      int result = -1;
      for(int i = 0; i < k; ++i){
        result = pq.top();
        pq.pop();
      }
      return result;
    }
};

int main(){
  vector<int> input = {3,2,1,5,6,4};
  Solution s;
  cout << s.findKthLargest(input, 2) << endl;
}